src/provider/store/Actions.ts

```diff
@@ -54,9 +54,9 @@ export class CreateNotification extends Action<RootState> {
         // Should only ever be at-most one notification with this ID, using filter over find as an additional sanity check
         const existingNotifications = store.state.notifications.filter(x => x.id === notification.id);
         if (existingNotifications.length) {
-            await new RemoveNotifications(existingNotifications).dispatch(store);
+            await (new RemoveNotifications(existingNotifications)).dispatch(store);
```

Comment:
With actions like this, where we dispatch another action, do we really want a delay between the synchronous reduce bit of one, and the synchronous reduce bit of the other? Feels risky, and makes it very hard to reason about what will happen if another action kicks off while this one is still running.

Maybe collect the promises and do `Promise.all`?

Reply:
> Maybe collect the promises and do `Promise.all`?

+1

Reply:
Changed to `Promise.all`.